### PR TITLE
Disable SSLv2 and SSLv3 in CurlHandler

### PR DESCRIPTION
**Pull request by stephentoub**
Forces CurlHandler to not use SSLv2 or SSLv3 even if they're available.  SSLv2 has been disabled by default in libcurl since 7.18.1, and SSLv3 since 7.39.0, but this ensures they're disabled even before those versions.

Fixes https://github.com/dotnet/corefx/issues/6668
cc: @bartonjs, @cipop, @davidsh, @kapilash, @joelverhagen

### CODE REVIEW

**Comment by blowdart** [comment]
:clap: That made my day

File: src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs [hunk 1]
```diff
@@ -25,6 +24,8 @@ private static class SslProvider
 
             internal static void SetSslOptions(EasyRequest easy, ClientCertificateOption clientCertOption)
             {
+                easy.SetCurlOption(Interop.Http.CURLoption.CURLOPT_SSLVERSION, (long)Interop.Http.CurlSslVersion.CURL_SSLVERSION_TLSv1); // disable SSLv2/SSLv3
```
Comment (bartonjs) [inline src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs]: I went to look this up in the documentation to confirm that this wasn't going to lock it to 1.0.

Might be worth a comment that says that this value enables all TLS 1.x versions and disables SSLv2 and SSLv3.

**Comment by bartonjs** [comment]
LGTM, aside from maybe adding one comment.

(Oh, no, what about the 0.01% (1/10000) of sites that still use SSLv3?  There, pretended to defend them.)

File: src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs (hunk 1 above)
Comment (justinvp) [inline src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs]: > I went to look this up in the documentation to confirm that this wasn't going to lock it to 1.0.

I had the same thought.

**Comment by justinvp** [comment]
:+1:

**Comment by davidsh** [comment]
LGTM

File: src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs]: Tweaked the comment that was there.

**Pull request merged**